pandas/tests/series/methods/test_value_counts.py

```diff
@@ -203,3 +203,22 @@ def test_value_counts_bool_with_nan(self, ser, dropna, exp):
         # GH32146
         out = ser.value_counts(dropna=dropna)
         tm.assert_series_equal(out, exp)
+
+    @pytest.mark.parametrize(
+        "input_array,expected",
+        [
+            (
+                [1 + 1j, 0, 1, 1j, 1 + 2j],
+                Series([1, 1, 1, 1, 1], index=[0, 1, 1j, 1 + 1j, 1 + 2j]),
```

Comment:
The order of keys is still arbitrary in value_counts, but it depends on the hash function used. #39009 should solve this problem. There are following options: a) having different number of each key, thus having an unique order b) if this PR will be merged prior to #39009, these tests must be fixed in #39009 c) if #39009 will be merged first, then these tests must be fixed in this PR. Option a) looks like safest approach.